iceoryx_posh/test/integrationtests/test_popo_port_user_building_blocks.cpp

```diff
@@ -0,0 +1,357 @@
+        maybeCaproMessage = publisherPortRouDi.getCaProMessage();
+
+        if (publisherThreadIndex == 0)
+        {
+            // First publisher thread will sync with subscriber
+            if (maybeCaproMessage.has_value())
+            {
+                caproMessage = maybeCaproMessage.value();
+                auto guardedVector = m_concurrentCaproMessageVector.GetScopeGuard();
+                guardedVector->push_back(caproMessage);
+            }
+
+            // Wait for subscriber to subscribe
+            caproMessage = waitForCaproMessage(m_concurrentCaproMessageVector, CaproMessageType::SUB);
+            m_caproMessageRx->push_back(caproMessage);
+
+            // Send ACK to subscriber
+            maybeCaproMessage = publisherPortRouDi.dispatchCaProMessage(m_caproMessageRx->back());
+            if (maybeCaproMessage.has_value())
+            {
+                caproMessage = maybeCaproMessage.value();
+                m_concurrentCaproMessageVector->push_back(caproMessage);
+            }
+        }
+        else
+        {
+            // All other publisher threads wait for the first thread to be synced with subscriber (indicated by
+            // receiving a SUB message) to continue
+            CaproMessage caproMessageRouDi(CaproMessageType::UNSUB, TEST_SERVICE_DESCRIPTION);
+            do
+            {
+                std::this_thread::sleep_for(std::chrono::microseconds(10));
+
+                if (m_caproMessageRx->size() != 0)
+                {
+                    caproMessageRouDi = m_caproMessageRx->back();
+                }
+
+            } while (caproMessageRouDi.m_type != CaproMessageType::SUB);
+
+            static_cast<void>(publisherPortRouDi.dispatchCaProMessage(caproMessageRouDi));
+        }
+
+        // Subscriber is ready to receive -> start sending samples
+        for (size_t i = 0; i < ITERATIONS; i++)
+        {
+            publisherPortUser.allocateChunk(sizeof(DummySample))
+                .and_then([&](ChunkHeader* chunkHeader) {
+                    auto sample = chunkHeader->payload();
+                    new (sample) DummySample();
+                    static_cast<DummySample*>(sample)->m_dummy = i;
+                    publisherPortUser.sendChunk(chunkHeader);
+                    m_sendCounter++;
+                })
+                .or_else([](AllocationError) {
+                    // Errors shall never occur
+                    FAIL();
+                });
+
+            /// Add some jitter to make thread breathe
+            std::this_thread::sleep_for(std::chrono::microseconds(rand() % 500));
+        }
+
+        // Signal the subscriber thread we're done
+        m_publisherRunFinished++;
+    }
+};
+
+TEST_F(PortUser_IntegrationTest, SingleProducer)
+{
+    std::thread subscribingThread(std::bind(&PortUser_IntegrationTest::subscriberThread<SubscriberPortSingleProducer>,
+                                            this,
+                                            1,
```

Comment:
Magic numbers are magic 🦄 What does this number here mean?

Reply:
Done.